chainer/distributions/geometric.py

```diff
@@ -0,0 +1,75 @@
+import chainer
+from chainer.backends import cuda
+from chainer import distribution
+from chainer.functions.math import ceil
+from chainer.functions.math import exponential
+
+
+class Geometric(distribution.Distribution):
+
+    """Geometric Distribution.
+
+    The probability density function of the distribution is expressed as
+
+    .. math::
+        f(x) = p(1-p)^{k-1},
+
+    Args:
+        p(:class:`~chainer.Variable` or :class:`numpy.ndarray` or \
+        :class:`cupy.ndarray`): Parameter of distribution.
+    """
+
+    def __init__(self, p):
+        super(Geometric, self).__init__()
+        self.__p = chainer.as_variable(p)
+
+    @property
+    def p(self):
+        return self.__p
+
+    @property
+    def batch_shape(self):
+        return self.p.shape
+
+    @property
+    def event_shape(self):
+        return ()
+
+    @property
+    def _is_gpu(self):
+        return isinstance(self.p.data, cuda.ndarray)
+
+    def log_prob(self, x):
+        return (x - 1) * exponential.log(1 - self.p) + exponential.log(self.p)
+
+    @property
+    def mean(self):
+        return 1 / self.p
+
+    def sample_n(self, n):
+        xp = cuda.get_array_module(self.p)
+        if xp is cuda.cupy:
+            eps = xp.random.uniform(
+                size=(n,)+self.batch_shape, dtype=self.p.dtype)
+        else:
+            eps = xp.random.uniform(
+                size=(n,)+self.batch_shape).astype(self.p.dtype)
+
+        noise = ceil.ceil(exponential.log(1 - eps)
+                          / exponential.log(1 - self.p))
```

Comment:
I don't think this is a reparameterized sampling.  Why don't you call `xp.random.geometric`?

Reply:
Oh, yes.